### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by llucax**
- Add missing enum members documentation
- Make cross-linking more consistent
- Improve Timer documentation

### CODE REVIEW

File: src/frequenz/channels/util/_select.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 
 class Selected(Generic[_T]):
-    """A result of a [`select`][frequenz.channels.util.select] iteration.
+    """A result of a [`select()`][frequenz.channels.util.select] iteration.
```
Comment (Marenz) [inline src/frequenz/channels/util/_select.py:21]: So we always want to use `()` when referencing functions?

File: src/frequenz/channels/util/_timer.py [hunk 2]
```diff
@@ -270,36 +270,49 @@ def __repr__(self) -> str:
 class Timer(Receiver[timedelta]):
     """A timer receiver that triggers every `interval` time.
 
-    The timer as microseconds resolution, so the `interval` must be at least
+    The timer as microseconds resolution, so the
```
Comment (Marenz) [inline src/frequenz/channels/util/_timer.py]: as -> has?

File: src/frequenz/channels/util/_timer.py [hunk 3]
```diff
@@ -270,36 +270,49 @@ def __repr__(self) -> str:
...
-    when it actually triggered.
+    The message it produces is a [`timedelta`][datetime.timedelta] containing the drift
+    of the timer, i.e. the difference between when the timer should have triggered and
+    the time when it actually triggered.
 
     This drift will likely never be `0`, because if there is a task that is
     running when it should trigger, the timer will be delayed. In this case the
     drift will be positive. A negative drift should be technically impossible,
-    as the timer uses `asyncio`s loop monotonic clock.
+    as the timer uses [`asyncio`][asyncio]s loop monotonic clock.
```
Comment (Marenz) [inline src/frequenz/channels/util/_timer.py:284]: We seem to be using a mix of of `()` and `[]` here. Is `[name][ref]` different from `(name)[ref]`?

File: src/frequenz/channels/util/_timer.py [hunk 4]
```diff
@@ -270,36 +270,49 @@ def __repr__(self) -> str:
...
 
-    If the timer is delayed too much, then the timer will behave according to
-    the `missed_tick_policy`. Missing ticks might or might not trigger
-    a message and the drift could be accumulated or not depending on the
-    chosen policy.
+    If the timer is delayed too much, then the timer will behave according to the
```
Comment (Marenz) [inline src/frequenz/channels/util/_timer.py]: > If the timer is delayed too much, then ~the timer~ **it** will

File: src/frequenz/channels/util/_select.py (hunk 1 above)
Comment (llucax) [reply src/frequenz/channels/util/_select.py:21]: In the rest of the file we always were referring to the select function as `select()`, so in this case I just wanted to have a consistent style.

File: src/frequenz/channels/util/_timer.py (hunk 3 above)
Comment (llucax) [reply src/frequenz/channels/util/_timer.py:284]: `[name](ref)` is a regular, standard, markdown link, `[name][ref]` is a `mkdocstrings` extension to do code-crosslinking, so `ref` can only be a valid fully-qualified Python symbol.

Comment (llucax) [reply src/frequenz/channels/util/_timer.py:284]: When configuring [external "object inventories"](https://mkdocstrings.github.io/python/usage/#import), this references can even point to external python packages, in this case `asyncio` will link to Python's official docs.

File: src/frequenz/channels/util/_timer.py (hunk 4 above)
Comment (llucax) [reply src/frequenz/channels/util/_timer.py]: Updated

File: src/frequenz/channels/util/_timer.py (hunk 2 above)
Comment (llucax) [reply src/frequenz/channels/util/_timer.py]: Updated

File: src/frequenz/channels/util/_timer.py (hunk 3 above)
Comment (llucax) [reply src/frequenz/channels/util/_timer.py:284]: BTW, `(name)[ref]` is incorrect, so if you spotted one is a bug, but I'm assuming is just a typo in your GitHub comment.

Comment (Marenz) [reply src/frequenz/channels/util/_timer.py:284]: This is one of my forever confusions in markdown syntax :sweat_smile: I can never remember the right order

**Review by Marenz — APPROVED** [review]

**Pull request merged**